### PR TITLE
TileMap: Add update_bitmask_region(2), expose update_bitmask_area(1) …

### PR DESCRIPTION
**Pull request by CyanBlob**
…and update_bitmask_region(2) to GDscript

### CODE REVIEW

**Comment by CyanBlob** [comment]
I went ahead and opened this PR since the changes seem rather harmless, while enabling the autotiling to be updated at runtime

**Comment by akien-mga** [comment]
CC @MarianoGnu for review.

**Comment by MarianoGnu** [comment]
I have 2 concerns related to usage only:
- If end.x < start.x or end.y < start.y then there will be no update, this is suposed to be the case or we should get maximum and minimum points and always update the area?
- If you update start to end bitmask tiles around that rectangle wont update, is this suposed to be the case or we should grow the rect 1 unit? (from start.x-1 to end.x +1 and from start.y-1 to end.y+1)

Other than that looks fine yo me 👍

**Comment by CyanBlob** [comment]
- I think it would be a good idea to always update the area. That should be easy enough to implement using get_used_cells (unless you meant to just invert start and end if necessary)

- Right now, update_bitmask_region calls update_bitmask_area, which does grow the rect. But now that I think about it, that will end up update_cell_bitmask(x, y) to be called multiple times per cell. I will update update_bitmask_region to grow the rect and call update_cell_bitmask(x, y) directly to remedy that.

I am at work now, but I will make these changes as soon as I get home!

**Comment by CyanBlob** [comment]
@MarianoGnu Have you had a chance to look at my updates?

File: scene/2d/tile_map.cpp [hunk 1]
```diff
@@ -1531,6 +1551,9 @@ void TileMap::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_recreate_quadrants"), &TileMap::_recreate_quadrants);
 	ClassDB::bind_method(D_METHOD("_update_dirty_quadrants"), &TileMap::_update_dirty_quadrants);
 
+	ClassDB::bind_method(D_METHOD("update_bitmask_area"), &TileMap::update_bitmask_area);
+	ClassDB::bind_method(D_METHOD("update_bitmask_region", "p_start", "p_end"), &TileMap::update_bitmask_region, DEFVAL(Vector2()), DEFVAL(Vector2()));
```
Comment (akien-mga) [inline scene/2d/tile_map.cpp]: For those bindings, you should remove the `p_` part of the parameters (they are used in the C++ signature, but not in the bindings for scripting APIs).

**Comment by CyanBlob** [comment]
@akien-mga Sorry about that! They're all squashed together now.

**Pull request merged**